### DISCUSSION TITLE
Can a live preview of Fill & Magic Wand tolerance be added?

### DISCUSSION THREAD

**Discussion by Agit1122** [PintaProject/Pinta#514] [Ideas]
When bucket-filling complex, antialiased areas with colors, it's very useful to preview it live, avoiding the need for undo & redo over and over until you get the tolerance right. This also applies to Magic Wand selections, etc. You can reach a fine snap-over point where a selection becomes clean and doesn't include stray pixels.

Windows Paint.NET has that seemingly rare feature, and I know someone with a **Mac** who could use it in freeware like Pinta.

Thanks.

**Comment by cameronwhite** [comment]
Yep, this is definitely something that more tools should support. Currently only the text tool and shape tools have "re-editing" functionality where you can keep adjusting the current operation instead of immediately finalizing it.

**Reply by Agit1122** [reply]
Even the fancy Krita paint program lacks live preview. In the freeware realm, I've only seen it in Paint.NET. Still unclear if Pinta actually shares code with that program or just uses it as a model.

**Comment by cameronwhite** [comment]
> Still unclear if Pinta actually shares code with that program or just uses it as a model.

Pinta has some code from the final version of Paint.NET that was open source (before v3.5), but things have diverged quite a bit in the many years since then and any new code is not shared

**Comment by RahlForge** [comment]
I am rezzing this thread, and will continue to rez it regularly. I came to Linux from Windows where I was using Paint.NET, which was my absolute go-to for image editing (over both GIMP and Photoshop... I _detest_ GIMP). When the Magic Wand is selected, and you click an area, then change the Tolerance, it _absolutely should automatically recalculate the selected area to match the new tolerance!_ This allowed me to very quickly adjust my selections to "shrink-wrap" to specific parts of an image. It's INCREDIBLY useful, because I don't have to pick, adjust, repick, adjust again, and repeat ad nauseum until I have the tolerance right. I'm using a lot of AI image generation stuff for some personal works, and the one thing the AI really sucks at so far is transparent backgrounds, so I have to do a lot of this kind of thing, and it was SUPER easy to do in Paint.NET, but is a bit clunky in Pinta.
